cpp/src/parquet/encoding.cc

```diff
@@ -941,6 +948,12 @@ class DictByteArrayDecoder : public DictDecoderImpl<ByteArrayType>,
     return result;
   }
 
+  int DecodeArrowNonNull(int num_values, ::arrow::BinaryDictionaryBuilder* out) override {
```

Comment:
I might be getting confused by the diff engine on GitHub, but is this the same code as on lines 725-730? If so, is there a common function both these methods could call?

Reply:
They are the same but I don't see an easy way to share an implementation.  Let me think on this one.

Reply:
I refactored this slightly in the latest commit.  @rdmello let me know how this looks to you.